components/odh-notebook-controller/controllers/notebook_controller_test.go

```diff
@@ -659,34 +672,51 @@ var _ = Describe("The Openshift Notebook controller", func() {
 			Expect(cli.Create(ctx, notebook)).Should(Succeed())
 			time.Sleep(interval)
 
+			By("By fetching the created Notebook for debugging")
+			actualNotebook := &nbv1.Notebook{}
+			key := types.NamespacedName{Name: Name, Namespace: Namespace}
+			Expect(cli.Get(ctx, key, actualNotebook)).Should(Succeed())
+
 			By("By checking that the webhook has injected the sidecar container")
-			Expect(*notebook).To(BeMatchingK8sResource(expectedNotebook, CompareNotebooks))
+			Expect(actualNotebook.Spec.Template.Spec.Volumes).To(ConsistOf(expectedNotebook.Spec.Template.Spec.Volumes))
```

Comment:
the modified assertion is no longer checking for sidecar container, right? so the `By` string would need to change too

Reply:
right, added a separated assertion  for the sidecar container

Reply:
thanks, looks ok to me now